op/model.py

```diff
@@ -91,6 +91,15 @@ def __init__(self, name, backend, cache):
     def __repr__(self):
         return f'<{type(self).__module__}.{type(self).__name__} {self.name}>'
 
+    @property
+    def is_leader(self):
+        if self.is_local:
+            # This value is not cached as it is not guaranteed to persist for the whole duration
+            # of a hook execution.
+            return self._backend.is_leader()
```

Comment:
given that it isnt cached, would it be better expressed as a method?
Also, it avoids having code think that it can
set
self.is_leader = True

having it be

if self.is_leader():

makes it slightly clearer that there is a query involved.

If it feels more natural as a property I can live with it. But they feel better as caching lazy loaded values to me.

Reply:
Done, thanks for the suggestion.